libs/micronaut-newrelic/src/main/java/com/agorapulse/micronaut/newrelic/BeanIntrospectionEventPayloadExtractor.java

```diff
@@ -44,10 +45,23 @@ public <E> Map<String, Object> extractPayload(E event) {
         Map<String, Object> map = new HashMap<>(propertyNames.length - 1);
 
         for (String name : propertyNames) {
-            introspection.getProperty(name)
-                .flatMap(p -> Optional.ofNullable(p.get(event)))
-                .map(v -> (v instanceof Boolean || v instanceof Number) ? v : String.valueOf(v))
-                .ifPresent(v -> map.put(name, v));
+            Optional<BeanProperty<E, Object>> property = introspection.getProperty(name);
```

Comment:
can you extract the body of the for each block to avoid using `continue`?

Reply:
not sure if that's what you meant but I removed the continues